docs/flask.asciidoc

```diff
@@ -0,0 +1,232 @@
+by raising an exception or logging a generic message.
+
+Capture an arbitrary exception by calling `capture_exception`:
+
+[source,python]
+----
+try:
+    1 / 0
+except ZeroDivisionError:
+    apm.capture_exception()
+----
+
+Log a generic message with `capture_message`:
+
+[source,python]
+----
+apm.capture_message('hello, world!')
+----
+
+[float]
+[[flask-logging]]
+=== Logging
+
+Passing `logging=True` to the `ElasticAPM` constructor will make the agent
+automatically log errors that are logging using Python's built-in `logging` module.
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+apm = ElasticAPM(app, logging=True)
+----
+
+For fine-grained control, you can initialize a logging handler and add it,
+just as you would any other handler.
+
+[source,python]
+----
+from flask import Flask
+
+from elasticapm.contrib.flask import ElasticAPM
+from elasticapm.handlers.logging import LoggingHandler
+
+app = Flask(__name__)
+
+apm = ElasticAPM(app)
+
+if __name__ == '__main__':
+    # Create a logging handler and attach it.
+    handler = LoggingHandler(client=apm.client)
+    handler.setLevel(logging.WARN)
+    app.logger.addHandler(handler)
+----
+
+You can also capture exceptions and send them explictly:
+
+[source,python]
+----
+@app.route('/')
+def bar():
+    try:
+        1 / 0
+    except ZeroDivisionError:
+        app.logger.error( 'I cannot math', exc_info=True)
+----
+
+NOTE: `exc_info=True` adds the exception info to the data that gets sent to the APM server.
+Without it, only the message is sent.
+
+[float]
+[[flask-extra-data]]
+==== Extra data
+
+If you want to send more data to ElasticAPM than what you get with the agent by default,
+logging can be done like so:
+
+[source,python]
+----
+@app.route('/')
+def bar():
+    try:
+        1 / 0
+    except ZeroDivisionError:
+        app.logger.error('Math is hard',
+            exc_info=True,
+            extra={
+                'good_at_math': False,
+            }
+        )
+    )
+----
+
+[float]
+[[flask-celery-tasks]]
+==== Celery tasks
+
+The Elastic APM agent will automatically send errors and performance data from your Celery tasks to the APM server.
+
+[float]
+[[flask-performance-metrics]]
+=== Performance Metrics
+
+If you've followed the instructions above, the agent has already hooked
+into the right signals and should be reporting performance metrics.
+
+[float]
+[[flask-ignoring-specific-views]]
+==== Ignoring specific routes
+
+You can use the `TRANSACTIONS_IGNORE_PATTERNS` configuration option to ignore specific routes.
+The list given should be a list of regular expressions which are matched against the transaction name:
+
+[source,python]
+----
+app.config['ELASTIC_APM'] = {
+    ...
+    'TRANSACTIONS_IGNORE_PATTERNS': ['^OPTIONS ', '/api/']
+    ...
+}
+----
+
+This would ignore any requests using the `OPTIONS` method
+and any `GET` requests containing `GET /api/`.
```

Comment:
why `GET`? 
`POST / api/` wouldn't be ignored?

Reply:
You're right of course!